lib/client.js

```diff
@@ -488,40 +497,58 @@ nodeUtil.inherits(Client, Raven);
 // todo these extra export props are sort of an adhoc mess, better way to manage?
 var defaultInstance = new Raven();
 defaultInstance.Client = Client;
-defaultInstance.patchGlobal = patchGlobal;
 defaultInstance.version = require('../package.json').version;
 defaultInstance.disableConsoleAlerts = utils.disableConsoleAlerts;
 
 module.exports = defaultInstance;
 
 function registerExceptionHandler(client, cb) {
-  var called = false;
+  // old cb sig: cb(didSend, errCaptured)
+  // now there is a cb 100%, no more if (cb)
+  var calledUncaughtHandler = false;
   process.on('uncaughtException', function (err) {
-    if (cb) { // bind event listeners only if a callback was supplied
-      var onLogged = function onLogged() {
-        called = false;
-        cb(true, err);
-      };
+    if (!calledUncaughtHandler) {
+      // this might be overly careful here - need to avoid getting messed up by recursion
+      calledUncaughtHandler = true;
+      var eventId = client.captureException(err, function (sendErr, eventId) {
+        try {
+          client.onFatalError(err);
+        } catch (e) {
+          // no-op just avoid inf loop...
+        }
+        // consider signature like client.onFatalError(err, sendErr, eventId);
```

Comment:
setOnFatalErrorCallback exists, yes, and it stores the callback in the `onFatalError` property - this comment is regarding the question "what should we pass to onFatalError when we invoke it"